node/heartbeat/sender.go

```diff
@@ -45,6 +47,7 @@ func NewSender(
 		privKey:       privKey,
 		marshalizer:   marshalizer,
 		topic:         topic,
+		shardID:       shardID,
 	}
 
 	return sender, nil
```

Comment:
line 61: should insert:
```GO
ShardID: s.shardCoordinator.SelfId(),
```
We can not start the heartbeat sender with a predefined shard ID as there is no guarantee that in time the node's shard ID is going to stay the same. That is why we need the shard coordinator instance that "knows" all the time the current time

Reply:
done